Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py

```diff
@@ -83,27 +87,67 @@ def editable(cls, branch, repository=None):
             return False
 
         remote_repo = repository.remote(name=repository.default_remote)
-        if remote_repo and isinstance(remote_repo, remote.GitHub):
-            for name in repository.source_remotes():
-                if branch in repository.branches_for(remote=name, cached=True):
-                    return False
+        if remote and remote_repo and isinstance(remote_repo, remote.GitHub):
+            return remote in repository.source_remotes()
         return True
 
     @classmethod
     def branch_point(cls, repository, limit=None):
         cnt = 0
-        commit = repository.commit(include_log=False, include_identifier=False)
-        while cls.editable(commit.branch, repository=repository):
-            cnt += 1
-            if limit and cnt > limit:
+        try:
+            proc = subprocess.Popen(
+                [repository.executable(), 'rev-list', 'HEAD'],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                cwd=repository.root_path,
+                **(dict(encoding='utf-8') if sys.version_info >= (3, 6) else dict())
+            )
```

Comment:
Can we add a comment making it clear why we're streaming this? (Presumably performance, given we're requesting the entire rev-list of HEAD.)